docs/operators/replace.md

```diff
@@ -0,0 +1,60 @@
+---
+title: replace
+category: Modify
+example: 'replace what=42, with=null'
+---
+
+Replaces all occurrences of a value with another value.
+
+```tql
+replace [path:field], what=any, with=any
+```
+
+## Description
+
+The `replace` operator scans all fields of each input event (optionally
+limited to a specific field) and replaces every occurrence of a value equal
+to `what` with the value specified by `with`. This is useful for normalizing,
+redacting, or cleaning up data in bulk.
+
+### `path: field (optional)`
+
+The field path to restrict substitution to. If omitted, all fields are considered.
+
+### `what: any`
+
+The value to search for and replace.
+
+### `with: any`
+
+The value to replace in place of `what`.
+
+## Examples
+
+### Replace all occurrences of 42 with null
+
+```tql
+replace what=42, with=null
+```
+
+### Replace all empty strings with "N/A" in the field `user.name`
+
+```tql
+replace user.name, what="" with="N/A"
+```
+
+### Replace a specific IP address with a redacted value
+
+```tql
+replace what=192.168.1.1, with="REDACTED"
```

Comment:
For this use case, what comes to mind immediately is a having a list of values. SPL has such a feature as well. There's a design question whether `list<T>` should match on `T`s or on the entire list when passed to `what`.

Reply:
Currently, it just matches the list itself. Let's see what users want.

Reply:
Okay, assuming you've checked this with similar other data tools.